gnovm/pkg/repl/repl_test.go

```diff
@@ -0,0 +1,137 @@
+package repl
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type step struct {
+	Line   string
+	Error  string
+	Result string
+}
+
+var fixtures = []struct {
+	Name      string
+	CodeSteps []step
+}{
+	{
+		Name: "Add new import",
+		CodeSteps: []step{
+			{
+				Line:   "import \"fmt\"\nimport \"os\"",
+				Result: "import \"fmt\"\nimport \"os\"\n",
+			},
+		},
+	},
+	{
+		Name: "Add wrong code",
+		CodeSteps: []step{
+			{
+				Line:  "importasdasd",
+				Error: "recovered from panic: test/test1.gno:7: name importasdasd not declared",
+			},
+			{
+				Line: "var a := 1",
+				// we cannot check the entire error because it is different depending on the used Go version.
+				Error: "error parsing code:",
+			},
+		},
+	},
+	{
+		Name: "Add function and use it",
+		CodeSteps: []step{
+			{
+				Line:   "func sum(a,b int)int{return a+b}",
+				Result: "func sum(a, b int) int\t{ return a + b }\n",
+			},
+			{
+				Line:   "import \"fmt\"",
+				Result: "import \"fmt\"\n",
+			},
+			{
+				Line:   "fmt.Println(sum(1,1))",
+				Result: "2\n",
+			},
+		},
+	},
+	{
+		Name: "All declarations at once",
+		CodeSteps: []step{
+			{
+				Line:   "import \"fmt\"\nfunc sum(a,b int)int{return a+b}",
+				Result: "import \"fmt\"\nfunc sum(a, b int) int\t{ return a + b }\n",
+			},
+			{
+				Line:   "fmt.Println(sum(1,1))",
+				Result: "2\n",
+			},
+		},
+	},
+	{
+		Name: "Fibonacci",
+		CodeSteps: []step{
+			{
+				Line: `
+				func fib(n int)int {
+					if n < 2 {
+						return n
+					}
+					return fib(n-2) + fib(n-1)
+				}
+				`,
+				Result: "func fib(n int) int {\n\tif n < 2 {\n\t\treturn n\n\t}\n\treturn fib(n-2) + fib(n-1)\n}\n",
+			},
+			{
+				Line:   "println(fib(24))",
+				Result: "46368\n",
+			},
+		},
+	},
+}
+
+func TestRepl(t *testing.T) {
+	for _, fix := range fixtures {
+		t.Run(fix.Name, func(t *testing.T) {
```

Comment:
`fix` variable is per loop, and needs to be unshared per iteration to avoid bugs when used in `t.Run()`, for example test parallelisation.
```suggestion
        fix := fix
		t.Run(fix.Name, func(t *testing.T) {
```